File: app/src/main/java/com/concrete/android/challenge/di/component/AppComponent.java
package com.concrete.android.challenge.di.component;

import android.app.Application;
import com.concrete.android.challenge.MyAppAplication;
import com.concrete.android.challenge.di.module.ActivityModule;
import com.concrete.android.challenge.di.module.AppModule;
import com.concrete.android.challenge.di.module.FragmentModule;
import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjectionModule;
import javax.inject.Singleton;

/**
 * @author Thiago Corredo
 * @since 2019-05-28
 */
@Singleton
@Component(modules = {
    AndroidInjectionModule.class, AppModule.class, ActivityModule.class, FragmentModule.class
})
public interface AppComponent {

  @Component.Builder
  interface Builder {

    @BindsInstance
    Builder application(Application application);

    AppComponent build();
  }

  void inject(MyAppAplication application);
}

File: app/src/main/java/com/rafaelpereiraramos/desafioAndroid/core/AppComponent.kt
package com.rafaelpereiraramos.desafioAndroid.core

import com.rafaelpereiraramos.desafioAndroid.App
import com.rafaelpereiraramos.desafioAndroid.di.ActivityContributorModule
import com.rafaelpereiraramos.desafioAndroid.di.AppModule
import com.rafaelpereiraramos.desafioAndroid.di.NetworkModule
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjectionModule
import javax.inject.Singleton

/**
 * Created by Rafael P. Ramos on 12/10/2018.
 */
@Singleton
@Component(modules = [AndroidInjectionModule::class, AppModule::class, NetworkModule::class, ActivityContributorModule::class])
interface AppComponent {
    @Component.Builder interface Builder {
        @BindsInstance
        fun application(application: App): Builder

        fun build(): AppComponent
    }

    fun inject(application: App)
}